FinalEngine.Tests/Editor/ViewModels/Dialogs/Layouts/ManageWindowLayoutsViewModelTests.cs

```diff
@@ -0,0 +1,284 @@
+// <copyright file="ManageWindowLayoutsViewModelTests.cs" company="Software Antics">
+// Copyright (c) Software Antics. All rights reserved.
+// </copyright>
+
+namespace FinalEngine.Tests.Editor.ViewModels.Dialogs.Layouts;
+
+using System;
+using System.Collections.Generic;
+using FinalEngine.Editor.ViewModels.Dialogs.Layout;
+using FinalEngine.Editor.ViewModels.Services.Actions;
+using FinalEngine.Editor.ViewModels.Services.Factories.Layout;
+using FinalEngine.Editor.ViewModels.Services.Layout;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+
+[TestFixture]
+public sealed class ManageWindowLayoutsViewModelTests
+{
+    private Mock<ILayoutManager> layoutManager;
+
+    private Mock<ILayoutManagerFactory> layoutManagerFactory;
+
+    private IList<string> layoutNames;
+
+    private Mock<ILogger<ManageWindowLayoutsViewModel>> logger;
+
+    private Mock<IUserActionRequester> userActionRequester;
+
+    private ManageWindowLayoutsViewModel viewModel;
+
+    [Test]
+    public void ApplyCommandCanExecuteShouldReturnFalseWhenSelectedItemIsNull()
+    {
+        // Arrange
+        this.viewModel.SelectedItem = null;
+
+        // Act
+        bool actual = this.viewModel.ApplyCommand.CanExecute(this.viewModel.SelectedItem);
+
+        // Assert
+        Assert.That(actual, Is.False);
+    }
+
+    [Test]
+    public void ApplyCommandCanExecuteShouldReturnTrueWhenSelectedItemNotNull()
+    {
+        // Arrange
+        this.viewModel.SelectedItem = "Test";
+
+        // Act
+        bool actual = this.viewModel.ApplyCommand.CanExecute(this.viewModel.SelectedItem);
+
+        // Assert
+        Assert.That(actual, Is.True);
+    }
+
+    [Test]
+    public void ApplyCommandExecuteShouldInvokeLayoutManagerLoadLayoutWhenInvoked()
+    {
+        // Arrange
+        this.viewModel.SelectedItem = "Layout 1";
+
+        // Act
+        this.viewModel.ApplyCommand.Execute(this.viewModel.SelectedItem);
+
+        // Assert
+        this.layoutManager.Verify(x => x.LoadLayout("Layout 1"));
+    }
+
+    [Test]
+    public void ConstructorShouldSetLayoutNamesToGetLayoutNamesWhenInvoked()
```

Comment:
> **Note** Preference only!!

Unit test method readability can be improved by following the convention:
`member_scenario_expectedresult` 

This makes it much easier to distinguish between the 3 important things that make up the test method to help understand what you are testing, the scenario under test, and the expected result of the test.

Sample:
```cs
    [Fact]
    public void TextureLoaded_WhenGettingValueAfterLoadingTexture_ReturnsTrue()
    {
        // Arrange
        var mockTexture = Substitute.For<IDisposable>();
        this.mockTextureLoader.LoadTexture(Arg.Any<string>()).Returns(mockTexture);
        var pool = CreatePool();

        // Act
        pool.LoadTexture();

        // Assert
        pool.TextureLoaded.Should().BeTrue();
    }
```

Reply:
You're probably right, this is much more readable and a common practice I have in other personal projects. I simply chose this convention when I started and continued with it. Perhaps I can address this alongside #238.